### PR TITLE
Jcf list / it-academy.by /  JC1-110-22 / Alena Trafimovich / Done tasks 1-2

### PR DESCRIPTION
**Pull request by AlenaTrafimovich**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/SchoolClass.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    };
+    public Comparator<Person> bySurnameName = new Comparator<>() {
+        @Override
+        public int compare(Person p1, Person p2) {
+            if (!p1.getSurname().equals(p2.getSurname())) {
+                return p1.getSurname().compareTo(p2.getSurname());
+            }
+            return p1.getName().compareTo(p2.getName());
+        }
+    };
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_list/SchoolClass.java:28]: move to separate classes

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/Studio.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    };
+    public Comparator<Actor> bySurnameAge = new Comparator<>() {
+        @Override
+        public int compare(Actor a1, Actor a2) {
+            if (!Objects.equals(a1.getLastName(), a2.getLastName())) {
+                return a1.getLastName().compareTo(a2.getLastName());
+            }
+            return Integer.compare(a1.getAge(), a2.getAge());
+        }
+    };
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_list/Studio.java:44]: move to separate classes

File: src/test/java/com/rakovets/course/java/core/practice/jcf_list/SchoolClassTest.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.Arrays;
+import java.util.List;
+
+public class SchoolClassTest {
+    private final Person person1 = new Person("Ivan", "Ivanov", 10, 3.0F);
+    private final Person person2 = new Person("Peter", "Petrov", 9, 3.0F);
+    private final Person person3 = new Person("Inna", "Innova", 10, 5.0F);
+    private final Person person4 = new Person("Irina", "Innova", 8, 4.0F);
+    private final SchoolClass schoolClass = new SchoolClass();
+    private final List<Person> list = new ArrayList<>(Arrays.asList(person1, person2, person3, person4));
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/jcf_list/SchoolClassTest.java:17]: move to init method

File: src/test/java/com/rakovets/course/java/core/practice/jcf_list/SchoolClassTest.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void testSortByAverageAnnualMark() {
+        schoolClass.getStudents().sort(schoolClass.byAverageAnnualMark);
+
+        List<Person> expected = Arrays.asList(person1, person2, person4, person3);
+
+        Assertions.assertEquals(expected, schoolClass.getStudents());
+    }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/jcf_list/SchoolClassTest.java:63]: fix

File: src/test/java/com/rakovets/course/java/core/practice/jcf_list/StudioTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    @Test
+    public void testSortByFee() {
+        studio.getActors().sort(studio.byFee);
+
+        List<Actor> expected = Arrays.asList(actor2, actor3, actor4, actor1);
+
+        Assertions.assertEquals(expected, studio.getActors());
+    }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/jcf_list/StudioTest.java:85]: fix

**Review by rakovets — CHANGES_REQUESTED** [review]
Need to fix

**Comment by AlenaTrafimovich** [comment]
I've fixed comments

**Pull request closed without merging**